code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -47,6 +49,21 @@
 		to_chat(user, "<span class='userdanger'>Your body is wracked with debilitating pain!</span>")
 		return
 
+	if(optional)

```

Comment:
this should probably be moved somewhere else; currently this can all go through, even if the stone's power is spent.